File: leetcode.com/1528_shuffle_string/solution.php
class Solution {
    /**
     * @param String $s
     * @param Integer[] $indices
     * @return String
     */
    function restoreString($s, $indices) {
        $r = $s;

        for ($i = 0 ; $i < strlen($s) ; $i++ ) {
            $r[$indices[$i]] = $s[$i];
        }

        return $r;
    }
}

File: leetcode.com/1528_shuffle_string/solution.c
char * restoreString(char *s, int *indices, int indicesSize)
{
    char *r = (char*) calloc(strlen(s) + 1, sizeof(char));

    for (int i = 0 ; i < indicesSize ; i++) 
        r[indices[i]] = s[i];

    return r;
}
